runtime/dart_isolate.cc

```diff
@@ -814,6 +819,19 @@ void DartIsolate::DartIsolateShutdownCallback(
                    << " exited with an error";
     FML_CHECK(tonic::LogIfError(sticky_error));
   }
+
+  // Terminate child Isolates
+  if ((*embedder_isolate)->terminate_child_isolates_) {
+    for (std::weak_ptr<DartIsolate> child_isolate :
+         (*embedder_isolate)->child_isolates_) {
+      if (!child_isolate.expired()) {
```

Comment:
You can call child_isolate.lock() here and then check whether the resulting shared_ptr is null